desktop/extensions-sdk/build/set-up/frontend-extension-tutorial.md

```diff
@@ -0,0 +1,309 @@
+    org.opencontainers.image.vendor="Docker Inc." \
+    com.docker.desktop.extension.api.version="0.3.0" \
+    com.docker.desktop.extension.icon="https://www.docker.com/wp-content/uploads/2022/03/Moby-logo.png"
+    com.docker.extension.screenshots="" \
+    com.docker.extension.detailed-description="" \
+    com.docker.extension.publisher-url="" \
+    com.docker.extension.additional-urls="" \
+    com.docker.extension.changelog=""
+
+COPY metadata.json .
+COPY docker.svg .
+COPY --from=client-builder /ui/build ui
+
+```
+
+  </div>
+  <div id="vue-dockerfile" class="tab-pane fade" markdown="1">
+
+<br/>
+
+> **Important**
+>
+> We don't have a working Dockerfile for Vue yet. [Fill out the form](https://docs.google.com/forms/d/e/1FAIpQLSdxJDGFJl5oJ06rG7uqtw1rsSBZpUhv_s9HHtw80cytkh2X-Q/viewform?usp=pp_url&entry.1333218187=Vue)
+> and let us know you'd like a Dockerfile for Vue.
+{: .important }
+
+  </div>
+  <div id="angular-dockerfile" class="tab-pane fade" markdown="1">
+
+<br/>
+
+> **Important**
+>
+> We don't have a working Dockerfile for Angular yet. [Fill out the form](https://docs.google.com/forms/d/e/1FAIpQLSdxJDGFJl5oJ06rG7uqtw1rsSBZpUhv_s9HHtw80cytkh2X-Q/viewform?usp=pp_url&entry.1333218187=Angular)
+> and let us know you'd like a Dockerfile for Angular.
+{: .important }
+
+  </div>
+  <div id="svelte-dockerfile" class="tab-pane fade" markdown="1">
+
+<br/>
+
+> **Important**
+>
+> We don't have a working Dockerfile for Svelte yet. [Fill out the form](https://docs.google.com/forms/d/e/1FAIpQLSdxJDGFJl5oJ06rG7uqtw1rsSBZpUhv_s9HHtw80cytkh2X-Q/viewform?usp=pp_url&entry.1333218187=Svelte)
+> and let us know you'd like a Dockerfile for Svelte.
+{: .important }
+
+  </div>
+</div>
+
+## Configure the metadata file
+
+A `metadata.json` file is required at the root of your extension directory.
+
+```json
+{
+  "icon": "docker.svg",
+  "ui": {
+    "dashboard-tab": {
+      "title": "UI Extension",
+      "root": "/ui",
+      "src": "index.html"
+    }
+  }
+}
+```
+
+## Use extension APIs in the application code
+
+To use the extension APIs and perform actions with Docker Desktop, the application must first import the 
+`@docker/extension-api-client` library. To install it, run the command below:
+
+```bash
+npm install @docker/extension-api-client
+```
+
+Then call the `createDockerDesktopClient` function to create a client object to call the extension APIs.
+
+```js
+import { createDockerDesktopClient } from '@docker/extension-api-client';
+
+const ddClient = createDockerDesktopClient();
+```
+
+When using Typescript, you can also install `@docker/extension-api-client-types` as a dev dependency. This will 
+provide you with type definitions for the extension APIs and auto-completion in your IDE.
+
+![types auto complete](images/types-autocomplete.png)
+
+For example, you can use the `docker.cli.exec` function to get the list of all the containers via the `docker ps` 
```

Comment:
```suggestion
For example, you can use the `docker.cli.exec` function to get the list of all the running containers via the `docker ps` 
```

Reply:
I updated the command instead as the code uses `--all` to list all the containers and not only the running ones.